addons/mail/static/src/rtc/voice_recorder.js

```diff
@@ -0,0 +1,51 @@
+/* @odoo-module */
+import { Component, useState, useRef } from "@odoo/owl";
+import { useService } from "@web/core/utils/hooks";
+import { useAttachmentUploader } from "@mail/attachments/attachment_uploader_hook";
+
+export class VoiceRecorder extends Component {
+    static props = ["thread", "composer?"];
```

Comment:
we should be able to get a thread from the composer I guess?

Reply:
Likely a typo, `composer` is mandatory but the `thread` is optional.
I kept the props but inverted the optional requirements.
Some improvements can be made to pass just a single prop: the attachment uploader of the composer. I will make a comment in following review, so that it's not in "outdated" comment like this one.